src/Livewire/Portal/Shop/Checkout.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace FluxErp\Livewire\Portal\Shop;
+
+use FluxErp\Actions\Comment\CreateComment;
+use FluxErp\Actions\Order\CreateOrder;
+use FluxErp\Actions\OrderPosition\CreateOrderPosition;
+use FluxErp\Livewire\Forms\AddressForm;
+use FluxErp\Models\Address;
+use FluxErp\Models\Order;
+use FluxErp\Models\OrderType;
+use FluxErp\Rulesets\Address\PostalAddressRuleset;
+use Illuminate\Validation\ValidationException;
+use Illuminate\View\View;
+use Livewire\Attributes\Renderless;
+use Livewire\Attributes\Rule;
+
+class Checkout extends Cart
+{
+    public ?string $comment = null;
+
+    public ?string $delivery_date = null;
+
+    public ?string $commission = null;
+
+    public AddressForm $address;
+
+    public AddressForm $deliveryAddress;
+
+    public bool $edit = true;
+
+    #[Rule('accepted')]
+    public bool $termsAndConditions = false;
+
+    public function mount(): void
+    {
+        $this->deliveryAddress->fill(auth()->user()->contact->deliveryAddress->toArray());
+    }
+
+    public function render(): View
+    {
+        return view('flux::livewire.portal.shop.checkout');
+    }
+
+    #[Renderless]
+    public function loadAddress(Address $address): void
+    {
+        $this->address->reset();
+        $this->address->fill($address->toArray());
+        $this->edit = ! $address->id;
+    }
+
+    #[Renderless]
+    public function loadTermsAndConditions(): string
+    {
+        return auth()->user()->contact->client->terms_and_conditions;
```

Comment:
if client has no terms_and_conditions (null), exception is thrown.

Reply:
√